File: jscomp/test/test_switch.js
'use strict';

var Curry = require("../../lib/js/curry.js");

function f(param) {
  if (typeof param !== "object") {
    if (param === "G") {
      return 4;
    } else {
      return 5;
    }
  }
  switch (param.TAG) {
    case "A" :
        return 0;
    case "B" :
        return 1;
    case "C" :
        return 2;
    case "F" :
        return 3;
    
  }
}

function bind(x, f) {
  if (x.TAG === "Left") {
    return {
            TAG: "Left",
            _0: Curry._1(f, x._0)
          };
  } else {
    return x;
  }
}

exports.f = f;
exports.bind = bind;
/* No side effect */

File: jscomp/test/test_switch.ml
type u = A of int | B of u | C of (int * int) | F of u * int | G | H

let f = function A _ -> 0 | B _ -> 1 | C _ -> 2 | F _ -> 3 | G -> 4 | H -> 5

type ('a, 'b) result = Left of 'a | Error of 'b

let bind x f = match x with Left x -> Left (f x) | Error _ as y -> y
